### PR TITLE
fix: Error message for cy.select() with no arguments

### PR DESCRIPTION
**Pull request by davidmunechika**
### User facing changelog
We include a more useful error message when `.select()` is  incorrectly called with no arguments.

### Additional details
Previous message: "Cannot read properties of undefined, (reading 'replace')"
image

New message: "`cy.select()` was not called with any value, index, or text"

Screenshot 2021-09-24 at 8 45 14 AM

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub?

### CODE REVIEW

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
This does fix the issue. Nice. 

This PR includes the files changes from the other feature though, adding the `index` feature. Can you have this PR only reflect the changes necessary for the error message when `cy.select()` has no arguments? 

If the PRs are dependent on each other, we could work on getting one in before the other. I see how the error message is a bit affected by the other PR.

**Comment by davidmunechika** [comment]
> This does fix the issue. Nice.
> 
> This PR includes the files changes from the other feature though, adding the `index` feature. Can you have this PR only reflect the changes necessary for the error message when `cy.select()` has no arguments?
> 
> If the PRs are dependent on each other, we could work on getting one in before the other. I see how the error message is a bit affected by the other PR.

@jennifer-shehane This should be updated now, thanks!

**Review by jennifer-shehane — APPROVED** [review]
Great! 💯

File: packages/driver/src/cy/commands/actions/select.ts [hunk 1]
```diff
@@ -11,6 +11,10 @@ const newLineRe = /\n/g
 export default (Commands, Cypress, cy) => {
   Commands.addAll({ prevSubject: 'element' }, {
     select (subject, valueOrTextOrIndex, options = {}) {
+      if (!valueOrTextOrIndex) {
```
Comment (chrisbreiding) [inline packages/driver/src/cy/commands/actions/select.ts:14]: Instead of just checking for existence, it would be better to validate that it's the right type. That's what we generally do with other command options. With this implementation, a valid option like `0` will incorrectly trigger the error while an invalid option like `true` will not.

Could you update it to ensure that it's a string or number, update the error message to make it clear those are the only types accepted, and then add a few tests for various valid/invalid values like `0`, `null`, `true`, `false`, `''` (empty string)?

**Review by chrisbreiding — CHANGES_REQUESTED** [review]

File: packages/driver/src/cypress/error_messages.ts [hunk 2]
```diff
@@ -1381,6 +1381,10 @@ export default {
   },
 
   select: {
+    no_argument: {
+      message: `${cmd('select')} was not called with any value, index, or text.`,
+      docsUrl: 'https://on.cypress.io/select',
+    },
```
Comment (chrisbreiding) [inline packages/driver/src/cypress/error_messages.ts:1387]: Let's update this to be called `invalid_argument` and make it similar to [this error message](https://github.com/cypress-io/cypress/blob/26f92e0e56d59cfa83207d101ad95683744f3696/packages/driver/src/cypress/error_messages.ts#L420-L423), so it details which types are support (string, number, or array) and shows what value the user passed in.

File: packages/driver/src/cy/commands/actions/select.ts [hunk 3]
```diff
@@ -11,6 +11,10 @@ const newLineRe = /\n/g
 export default (Commands, Cypress, cy) => {
   Commands.addAll({ prevSubject: 'element' }, {
     select (subject, valueOrTextOrIndex, options = {}) {
+      if (!_.isNumber(valueOrTextOrIndex) && !_.isString(valueOrTextOrIndex) && !_.isArray(valueOrTextOrIndex)) {
+        $errUtils.throwErrByPath('select.no_argument')
+      }
+
```
Comment (chrisbreiding) [inline packages/driver/src/cy/commands/actions/select.ts:17]: If we detect the value is an array, we should look at its values and validate them as strings or numbers as well. It probably needs a separate error message, since arrays can't be nested inside.

**Review by chrisbreiding — CHANGES_REQUESTED** [review]

File: packages/driver/src/cy/commands/actions/select.ts (hunk 1 above)
Comment (jennifer-shehane) [reply packages/driver/src/cy/commands/actions/select.ts]: Nice catch Chris.

**Review by chrisbreiding — APPROVED** [review]
